openlr/openlr_model.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "geometry/latlon.hpp"
+#include "geometry/point2d.hpp"
+
+#include "std/limits.hpp"
+
+namespace openlr
+{
+enum class FormOfAWay
+{
+  // The physical road type is unknown.
+  Undefined,
+  // A road permitted for motorized vehicles only in combination with a prescribed minimum speed.
+  // It has two or more physically separated carriageways and no single level-crossings.
+  Motorway,
+  // A road with physically separated carriageways regardless of the number of lanes.
+  // If a road is also a motorway, it should be coded as such and not as a multiple carriageway.
+  MultipleCarriageway,
+  // All roads without separate carriageways are considered as roads with a single carriageway.
+  SingleCarriageway,
+  // A road which forms a ring on which traffic traveling in only one direction is allowed.
+  Roundabout,
+  // An open area (partly) enclosed by roads which is used for non-traffic purposes
+  // and which is not a Roundabout.
+  Trafficsquare,
+  // A road especially designed to enter or leave a line.
+  Sliproad,
+  // The physical road type is known, but does not fit into one of the other categories.
+  Other,
+  // A path only allowed for bikes.
+  BikePath,
```

Comment:
This version of OpenLR is modified by traffic provider. It's not the vanilla version.